### PR TITLE
Completion for exports

### PR DESCRIPTION
**Pull request by vladima**
Commits 'replace -1 in SmartIndenter with const enum' and 'dded missing declaration kinds to getMeaningForDeclaration' are not related to the primary idea of this PR - it was fixes for asserts that appear because of malformed code

### CODE REVIEW

**Comment by JsonFreeman** [comment]
:+1: once all comments are addressed

File: src/services/formatting/smartIndenter.ts [hunk 1]
```diff
@@ -2,6 +2,11 @@
 
 module ts.formatting {
     export module SmartIndenter {
+
+        const enum Value {
```
Comment (mhegazy) [inline src/services/formatting/smartIndenter.ts]: Possibly `export const Unkown = -1;` after your let/const PR gets in

File: src/services/services.ts [hunk 2]
```diff
@@ -2403,6 +2403,19 @@ module ts {
                         getCompletionEntriesFromSymbols(filteredMembers, activeCompletionSession);
                     }
                 }
+                else if (getAncestor(previousToken, SyntaxKind.ImportClause)) {
+                    // cursor is in import clause
+                    // try to show exported member for imported module
+                    isMemberCompletion = true;
+                    isNewIdentifierLocation = true;
```
Comment (mhegazy) [inline src/services/services.ts]: Shouldn't this be `previousToken.kind === SyntaxKind.AsKeyword`

Comment (JsonFreeman) [inline src/services/services.ts]: Also, shouldn't isMemberCompletion be `previousToken.kind !== SyntaxKind.AsKeyword`?

File: src/services/formatting/smartIndenter.ts (hunk 1 above)
Comment (JsonFreeman) [inline src/services/formatting/smartIndenter.ts]: Oh the PR inlines consts?

Comment (vladima) [inline src/services/formatting/smartIndenter.ts]: no, we don't inline consts (though we can do this in some cases)

**Pull request merged**